File: app/src/main/java/com/example/josycom/flowoverstack/viewmodel/QuestionViewModel.java
package com.example.josycom.flowoverstack.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;
import androidx.paging.PagedList;

import com.example.josycom.flowoverstack.model.Question;
import com.example.josycom.flowoverstack.model.QuestionDataSource;
import com.example.josycom.flowoverstack.model.QuestionDataSourceFactory;
import com.example.josycom.flowoverstack.repository.QuestionRepository;

import java.util.Objects;

public class QuestionViewModel extends ViewModel {

    private LiveData<PagedList<Question>> mQuestionPagedList;
    private LiveData<String> networkState;

    QuestionViewModel(int page, int pageSize, String order, String sortCondition, String site, String filter) {
        QuestionRepository questionRepository = new QuestionRepository(page, pageSize, order, sortCondition, site, filter);
        mQuestionPagedList = questionRepository.getQuestionPagedList();
        LiveData<QuestionDataSource> liveDataSource = QuestionDataSourceFactory.getQuestionLiveDataSource();
        networkState = Transformations.switchMap(liveDataSource, QuestionDataSource::getNetworkState);
    }

    public LiveData<PagedList<Question>> getQuestionPagedList() {
        return mQuestionPagedList;
    }

    public LiveData<String> getNetworkState() {
        return networkState;
    }

    public void refresh() {
        Objects.requireNonNull(QuestionDataSourceFactory.getQuestionLiveDataSource().getValue()).invalidate();
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/viewmodel/QuestionViewModel.kt
package com.josycom.mayorjay.flowoverstack.ui.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import androidx.paging.PagingData
import androidx.paging.cachedIn
import com.josycom.mayorjay.flowoverstack.model.Question
import com.josycom.mayorjay.flowoverstack.data.QuestionRepository
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class QuestionViewModel @Inject constructor(questionRepository: QuestionRepository, page: Int, pageSize: Int, order: String, sortCondition: String, site: String, tagged: String, filter: String, siteKey: String) : ViewModel() {

    val questionDataFlow: Flow<PagingData<Question>>?

    init {
        questionRepository.init(page, pageSize, order, sortCondition, site, tagged, filter, siteKey)
        questionDataFlow = questionRepository.questionDataFlow?.cachedIn(viewModelScope)
    }
}